Modules/System/Convert/src/ConvertImpl.Codeunit.al

```diff
@@ -0,0 +1,25 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 1486 "Convert Impl."
+{
+    Access = Internal;
+
+    procedure ConvertTextEncoding(SrcCodepage: Integer; DstCodepage: Integer; Text: Text) ConvertedText: Text
+    var
+        Encoding: DotNet Encoding;
+        SrcEncoding: DotNet Encoding;
+        DstEncoding: DotNet Encoding;
+        SrcBytes: DotNet Array;
+        DstBytes: DotNet Array;
+    begin
+        SrcEncoding := Encoding.GetEncoding(SrcCodepage);
```

Comment:
Parameter validation is needed here :)

What if `SrcCodepage` is -1?

Reply:
If it is -1 than an error appears: 
![image](https://user-images.githubusercontent.com/18716238/105042155-2f327f00-5a64-11eb-89c1-8c3fc0aaadab.png)

Isn't that what you would expect? If you would exit the code you won't know that the text is not encoded...